model/host/host.go

```diff
@@ -989,7 +989,7 @@ func (h *Host) UpdateRunningTask(t *task.Task) (bool, error) {
 
 	selector := bson.M{
 		IdKey:          h.Id,
-		StatusKey:      evergreen.HostRunning,
+		StatusKey:      bson.M{"$in": []string{evergreen.HostProvisioning, evergreen.HostRunning}},
```

Comment:
Do we also want to allow HostStarting here? I think when I was investigating some hosts were in this state and asking for tasks

Reply:
👍 isn't this the main issue (given that hosts don't go into provisioning until DescribeInstances tells us they're running)?

Reply:
Done.

Reply:
Do we want to allow all those statuses though? I can't think of how we might get into the situation, but it seems weird that we'd let an uninitialized host run a task

Reply:
I only did this to be consistent with Sam's original PR. However, I don't really want to allow other statuses so I'm going to ban all statuses except starting, provisioning, and running and only for the user data case.